### ISSUE TITLE
Consider `client` type renames

### ISSUE THREAD

**Issue by carllerche** [tower-rs/tower-h2#2]
Rename `client::Service` -> `client::Connection`. This is the actual, established, connection. It is backed by a single `TcpStream` and implements `Service`. Another option is to name this `Client`.

Then, the simple case to establish a connection would be: `Connection::connect(&addr)` (or `Client::connect(&addr)`.

`client::Client` -> client::Connect`. this implements `tokio_connect::Connect` and is essentially the connection factory.

**Comment by seanmonstar** [comment]
And completely remove any type named `Client`?

**Comment by carllerche** [comment]
@seanmonstar for now, yes. I think a Client type would be best reserved for something that establishes a configured connection with a socket address or host name (dns resolution)

Basically, reserve Client for later.

**Comment by olix0r** [comment]
I agree with the rename.

I have some thoughts on things we might want in a proper Client interface.  Something like finagle's StackClient may not be particularly well-suited for Rust; but we may find some nice middle-ground with macros, etc...  Finagle started with a ClientBuilder type, but this type became overly complex and not generic across protocols.  Eventually, we mostly replaced the builder types with _Stack_-based types that support:

- a generic type-map for configuration

- a list of, effectively:
```
trait Module {
  type Request;
  type Response;
  type Error;
  type NewService: NewService<Request = Self::Request, Response = Self::Response, Error = Self::Error>;

  fn make<N>(params: Params, next: N) -> Self::NewService
      where N: Module<Request = Self::Request, Response = Self::Response, Error = Self::Error>;
}
```

i know this is nowhere near compiling, just illustrative...

Anyway, even if we don't end up with stacks, clients should try to be generic over discovery, stream classification (success/failures), retry policy, load balancing, sharding, etc etc

**Comment by carllerche** [comment]
This issue has been resolved by #4.

There is also #6, which adds `Connection::handshake`. To add a `connect` fn would require depending on tokio-core, which I would like to punt for now.
